File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/util/logging/ConsoleLogger.java
package by.bntu.fitr.povt.cng.model.util.logging;

import by.bntu.fitr.povt.cng.model.util.logging.Decorator.TextDecorator;

import java.util.Date;

public class ConsoleLogger extends AbstractLogger implements Logger {

    private TextDecorator decorator;

    public ConsoleLogger() {
    }

    public ConsoleLogger(TextDecorator decorator) {
        this.decorator = decorator;
    }

    @Override
    public void logEvent(String string) {
        var baseString = super.getBaseString();
        if (decorator != null) {
            string = decorator.change(string);
        }
        System.out.println(baseString + string);
    }
}

File: Kotlin/TryKotlin/src/by/bntu/fitr/povt/cng/model/loggers/ConsoleLogger.kt
package by.bntu.fitr.povt.cng.model.loggers

import by.bntu.fitr.povt.cng.model.decorators.CaseDecorator

class ConsoleLogger : AbstractLogger,  Logger {

    private var decorator: CaseDecorator?

    constructor(){
        decorator = null
    }

    constructor(decorator : CaseDecorator){
        this.decorator = decorator
    }

    override fun logEvent(event: String) {
        val base: String = super.getDate()
        if (decorator != null){
            println(decorator!!.decorate("$base:\t $event"))
        }else{
            println("$base:\t $event")
        }


    }
}